vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -651,3 +669,27 @@ func (s *GenericAPIServer) Eventf(eventType, reason, messageFmt string, args ...
 		klog.Warningf("failed to create event %s/%s: %v", e.Namespace, e.Name, err)
 	}
 }
+
+// terminationLoggingListener wraps the given listener and logs new connections
+// after the stopCh has been closed, i.e. when termination has begun.
+type terminationLoggingListener struct {
+	net.Listener
+	stopCh, lateStopCh <-chan struct{}
+}
+
+func (l *terminationLoggingListener) Accept() (net.Conn, error) {
+	c, err := l.Listener.Accept()
+	if err != nil {
+		return nil, err
+	}
+
+	select {
```

Comment:
mhm, not deterministic if both are true `lateStopCh`and `stopCh`

Reply:
it is. the first select only consders lateStopCh and returns.

Reply:
done

Reply:
now it is before it wasn't